marimo/_islands/island_generator.py

```diff
@@ -220,10 +239,24 @@ def render_head(
                 integrity="sha384-wcIxkf4k558AjM3Yz3BBFQUbk/zgIYC2R0QpeeYb+TwlBVMrlgLqwRjRtGZiK7ww"
                 crossorigin="anonymous"
             />
+
+            <marimo-version data-version="{version_override}" hidden="">
+            </marimo-version>
+            <marimo-mode data-mode="island" hidden=""></marimo-mode>
+
+            <!--
+              Required to silence marimo expectations, but not actively used
+              right now.
+            -->
+            <marimo-user-config data-config="{{}}" hidden="">
+            </marimo-user-config>
+            <marimo-app-config data-config="{{}}"> </marimo-app-config>
```

Comment:
i think its a bug if a codepath is using this / expecting this. if you remove this, does it still fail?

It should be handled by:
```
const {
  getMarimoVersion,
  getMarimoServerToken,
  getMarimoAppConfig,
  getMarimoUserConfig,
  getMarimoCode,
} = import.meta.env.VITE_MARIMO_ISLANDS
  ? islandsBasedMarimoSettings
  : domBasedMarimoSettings;
```

Reply:
Doesn't fail, but does produce error in console

Reply:
ah got it , thanks for explaining. PR looks good, thank you for the fixes!

Reply:
Woops, I missed there as well new commit should check for islands from mode.

So now normal marimo builds work for islands, separate islands builds aren't needed

Reply:
Ah nice - that's cool actually